### PR TITLE
My React-redux-bookstore-app Milestone 2

### PR DESCRIPTION
**Pull request by Davidosky007**
# My React-redux-bookstore-app Milestone 2
In this milestone i
* Created a components directory in the src folder for all the presentational components.
* Created a containers directory in the src folder for all the container components.
* Created an actions directory with the index.js file where i defined all the Redux actions.
* Created a reducers directory where i have all my reducer functions
* Implemented the BooksList component which displays the books as an html table
* Implemented the BooksForm component which renders a simple form to add new books
* Implemented the Book component which represents a single book
* Initialized the Redux store into my app using <Provider>

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @Davidosky007  👋

**You've done pretty well here, BookForm, reducers, and actions well written. 👌**

However, let's make a couple of changes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/components/Book.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    category: PropTypes.string.isRequired,
+  })).isRequired,
+  handleRemoveBook: PropTypes.func.isRequired,
+};
+
+const mapDispatchToProps = (dispatch) => ({
+  removeBook: (id) => dispatch(removeBook({ id })),
+});
+
+export default connect(null, mapDispatchToProps)(Book);
```
Comment (mikenath223) [inline src/components/Book.js]: - [ ] This component shouldn't be connected to the store. Kindly remove this. Essentially child components shouldn't be connected to the store only connect parent components.

File: src/containers/BookList.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        <tr>
+          <th>BOOK ID</th>
+          <th>Title</th>
+          <th>Category</th>
+        </tr>
+      </thead>
+      <tbody>
+        {books && books.length
+          ? books.map((book) => <Book key={`book-${book.id}`} book={book} handleRemoveBook={() => removeBook(book)} />)
+          : alert('No booklist to show!')}
```
Comment (mikenath223) [inline src/containers/BookList.js]: - [ ] Please remove this alert statemen, as it blocks running code.

File: src/components/Book.js (hunk 1 above)
Comment (Davidosky007) [reply src/components/Book.js]: Alright thanks

File: src/index.js [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
+import { createBook } from './actions/index';
 import './index.css';
-import App from './App';
 
+const store = configureStore();
+const categories = ['Action', 'Horror', 'History', 'Kids', 'Biography', 'Learning', 'Sci-Fi'];
+store.dispatch(createBook({ id: Math.floor(Math.random() * 100), title: 'Harry Potter', category: categories[6] }));
+store.dispatch(createBook({ id: Math.floor(Math.random() * 100), title: 'Rich Kid Poor Dad', category: categories[4] }));
+store.dispatch(createBook({ id: Math.floor(Math.random() * 100), title: 'How To Code In One Week', category: categories[2] }));
+store.dispatch(createBook({ id: Math.floor(Math.random() * 100), title: 'Do Not Use Git Rebase', category: categories[1] }));
```
Comment (mikenath223) [inline src/index.js]: - [ ] Also kindly remove this as this isn't the proper way to set the initial state. InitialState should be set when creating the reducer:
```
const initialState = [ array of book objects];

const booksReducer = (state = initialState, action) => {
...
}
```

Comment (Davidosky007) [reply src/index.js]: alright thanks

File: src/containers/BookList.js (hunk 2 above)
Comment (Davidosky007) [reply src/containers/BookList.js]: alright thanks

**Review by Berabjesus — APPROVED** [review]
# **Status: PROJECT APPROVED** 
#### Great job implementing the changes required @Davidosky007 
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete. 👍** 

I wish you all the very best in your next tasks:1st_place_medal:
#### **If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help** 
**HAPPY CODING :computer:[Bereket Beshane](https://bereketababu.me)** | [LinkedIn](https://linkedin.com/in/bereketbeshane) | [GitHub](https://github.com/Berabjesus)

**Pull request merged**